core/src/test/java/org/elasticsearch/cluster/NoMasterNodeIT.java

```diff
@@ -121,23 +121,21 @@ public void run() {
                 ClusterBlockException.class, RestStatus.SERVICE_UNAVAILABLE
         );
 
-        checkWriteAction(
-                false, timeout,
+        checkWriteAction(timeout,
                 client().prepareUpdate("test", "type1", "1")
                         .setScript(new Script(
                             ScriptType.INLINE, Script.DEFAULT_SCRIPT_LANG, "test script", Collections.emptyMap())).setTimeout(timeout));
 
-        checkWriteAction(
-                autoCreateIndex, timeout,
+        checkWriteAction(timeout,
                 client().prepareUpdate("no_index", "type1", "1")
                         .setScript(new Script(
                             ScriptType.INLINE, Script.DEFAULT_SCRIPT_LANG, "test script", Collections.emptyMap())).setTimeout(timeout));
 
 
-        checkWriteAction(false, timeout,
+        checkWriteAction(timeout,
                 client().prepareIndex("test", "type1", "1").setSource(XContentFactory.jsonBuilder().startObject().endObject()).setTimeout(timeout));
 
-        checkWriteAction(autoCreateIndex, timeout,
```

Comment:
This is actually due to `TransportUpdateAction` behaving differently (on how it handles cluster blocks when `autoCreateIndex` is `true`, instead of throwing cluster block exception, transport update throws master not discovered exception). IMO, update action should be fixed to be consistent here and throw a cluster block exception instead. I re-worked the tests, so the distinction is clear.